docs/pages/upgrading/self-hosted-kubernetes.mdx

```diff
@@ -0,0 +1,153 @@
+---
+title: Upgrade Self-Hosted Teleport Clusters on Kubernetes
+description: Provides instructions for upgrading self-hosted Teleport clusters that run on Kubernetes. 
+---
+
+This guide explains how to upgrade self-hosted Teleport clusters running on
+Kubernetes.
+
+## Prerequisites
+
+- Familiarity with the [Upgrading Compatibility Overview](./overview.mdx) guide,
+  which describes the sequence in which to upgrade components of your cluster.
+- A self-hosted Teleport cluster in which the Auth Service and Proxy Service run
+  on Kubernetes. This guide assumes that you have deployed the Teleport cluster
+  using the `teleport-cluster` Helm chart.
+
+Teleport supports automatic agent updates for the `teleport-kube-agent` Helm
+chart. The [Automatic Update Architecture
+](../architecture/agent-update-management.mdx) guide describes how agent
+updating works. Automatic agent upgrades require:
+
+- A Teleport cluster running version 13.0.0 or above.
+- At least one Teleport Enterprise agent deployed via the `teleport-kube-agent`
+  Helm chart.
+- A **version server** in your infrastructure that agents can query in order to
+  upgrade themselves. Read [Set up Automatic Upgrade
+  Infrastructure](./self-hosted-automatic-agent-updates.mdx) for how to deploy a
+  version server.
+
+This guide assumes that you have configured the `teleport-cluster` Helm chart
+with a values file called `values.yaml`, and that your `teleport-cluster`
+release is called `teleport-cluster`.
+
+## Step 1/3. Shrink the Auth Service pool
+
+You must reduce the number of Auth Service instances to one in order to ensure a
+consistent cluster state during the upgrade.
+
+Ensure that your `teleport-cluster` values file includes the followinmg
+configuration:
+
+```yaml
+auth:
+  highAvailability:
+    replicaCount: 1
+```
+
+Once you have completed this guide and upgraded the cluster, you can configure
+your cluster for high availability again.
+
+## Step 2/3. Upgrade the Auth Service and Proxy Service
+
+Run the following commands to upgrade Auth Service and Proxy Service instances
+running on Kubernetes.
+
+1. Update the Teleport Helm chart repository so you can install the latest
+   version of the `teleport-cluster` chart:
+
+    (!docs/pages//kubernetes-access/helm/includes/helm-repo-add.mdx!)
+
+1. Upgrade the Helm release:
+
+    ```code
+    $ helm upgrade teleport-cluster teleport/teleport-cluster --values=values.yaml
```

Comment:
Regarding the helm upgrade commands (comments valid for all `helm` upgrade guides):
- `helm repo update` should be run before (because if we don't specify it we'll definitely have support tickets about folks not seeing the new release)
- Choosing the version with the `--version` flag.
  ```shell
  helm upgrade <release-name> teleport/teleport-cluster --version <THE VERSION YOU WANT>
  ```
  If we don't explicitly set this we'll end up with 2 issues:
  - users installing the wrong version (surprise downgrade, or 2+ major upgrades)
  - users using the `teleportVersionOverride` Helm chart value to pin the version (we really don't want this to happen, this is a footgun)
- nit: it might be nice to make the release name a variable, because a lot of customers changed it and it's not necessarily clear which part of the command is the release name and should be changed.